test/e2e/common_test.go

```diff
@@ -138,10 +139,29 @@ const (
 	imageCacheDir = "imagecachedir"
 )
 
+var netnsFiles []fs.DirEntry
+
+func getNetnsDir() string {
+	if isRootless() {
+		if env, ok := os.LookupEnv("XDG_RUNTIME_DIR"); ok {
+			return filepath.Join(env, "netns")
+		}
+	} else {
+		return "/run/netns"
+	}
+	return ""
+}
+
 var _ = SynchronizedBeforeSuite(func() []byte {
 	globalTmpDir, err := os.MkdirTemp("", "podman-e2e-")
 	Expect(err).ToNot(HaveOccurred())
 
+	netnsFiles, err = os.ReadDir(getNetnsDir())
```

Comment:
This relies on (undocumented, or at least unclear) behavior of `os.ReadDir("")` yielding `ErrNotExist`. Someone who does not know that (like me) could reasonably believe that `os.Readdir("")` would interpret `""` as `.`, and that this test would be operating on cwd in the XDG-unset case, and spend time looking at `os.ReadDir()` to see what it does.

One solution might be to have `getNetnsDir()` never return empty; another might be for it to throw an error if XDG is unset.

Reply:
Yeah I went with the approach of less lines. Things like `ls ""`, `stat ""` always return ENOENT so I would not call this unclear behavior. But yeah sure go could behave differently here and there are certainly other cases where it does some special things.

What do you mean by never return empty? Should I just return some bogus "Idonotexists" string instead?

Reply:
Just that if you use a default XDG value, the function will never return empty

Reply:
ok will do

Reply:
>Things like ls "", stat "" always return ENOENT

Not really the same, because of shell quoting weirdness. I would consider `os.ReadDir("")` to be equivalent to `ls` with no arguments. Anyhow, moot point if you use a default XDG